src/main/java/de/bonndan/nivio/search/SearchDocumentFactory.java

```diff
@@ -48,7 +49,7 @@ public static Document from(Item item) {
         document.add(new TextField(LUCENE_FIELD_COMPONENT_TYPE, "item", Field.Store.YES));
         document.add(new TextField(LUCENE_FIELD_FQI, item.getFullyQualifiedIdentifier().toString(), Field.Store.YES));
 
-        document.add(new TextField(LUCENE_FIELD_IDENTIFIER, Optional.ofNullable(item.getIdentifier()).orElse(""), Field.Store.YES));
+        document.add(new TextField(LUCENE_FIELD_IDENTIFIER, item.getIdentifier(), Field.Store.YES));
         document.add(new TextField(LUCENE_FIELD_NAME, Optional.ofNullable(item.getName()).orElse(""), Field.Store.YES));
```

Comment:
The IDE tells me, that `item.getIdentifier()` cannot be null. However, you said, it can be.
Either remove the annotation, or enforce, that @NonNull is true.

Generally: Using @NonNull and @Nullable can be very helpful. However, if not used consistently, it gives the developer a false sense of security.

I would have assumed, my IDE is right and I would have applied the suggested change.

Reply:
done